### ISSUE TITLE
UI Tests that check ICEing thread names break with `parallel-compiler = true`

### ISSUE THREAD

**Issue by anp** [rust-lang/rust#65047]
When I run `./x.py test` locally on a clean checkout of master with `parallel-compiler = true`, I see a couple of failures that seem to be related to the name of the rustc thread which ICEd in the tests. I turned parallel-compiler off, reran the test command, and the tests passed.

Here's the stderr diffs of the failing tests:

```
---- [ui] ui/consts/miri_unleashed/mutable_references_ice.rs stdout ----
diff of stderr:

4	LL |     x: &UnsafeCell::new(42),
5	   |        ^^^^^^^^^^^^^^^^^^^^
6	
-	thread 'rustc' panicked at 'assertion failed: `(left != right)`
+	thread '<unnamed>' panicked at 'assertion failed: `(left != right)`
8	  left: `Const`,

...

---- [ui] ui/pattern/const-pat-ice.rs stdout ----
diff of stderr:

-	thread 'rustc' panicked at 'assertion failed: rows.iter().all(|r| r.len() == v.len())', src/librustc_mir/hair/pattern/_match.rs:LL:CC
+	thread '<unnamed>' panicked at 'assertion failed: rows.iter().all(|r| r.len() == v.len())', src/librustc_mir/hair/pattern/_match.rs:LL:CC
2	note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

My assumption is that the parallel-compiler flag doesn't result in all of the threads being named `rustc`, and so when the ICE occurs it produces an error message from an unnamed thread. I also assume this is totally OK in most cases, because very few things rely on the names of compiler threads.

If these completely unfounded assumptions are accurate, then would it be reasonable to set the name of threads in the compiler's pool?

* Commit: c6293e359848c8c7275330719b33ccd168df04c3
* OS: Arch Linux (fairly current as of filing)

**Comment by pnkfelix** [comment]
triage: P-high, since fixing this is presumably high impact for WG-parallel-rustc. Removing nomination label.

**Comment by mati865** [comment]
Would it be enough to normalize output of those two tests similarly to this? https://github.com/rust-lang/rust/blob/0221e265621a5fcc68ca62bdcdeabad1882a0e9a/src/test/ui/consts/miri_unleashed/mutable_references_ice.rs#L4

**Comment by pnkfelix** [comment]
(also, I'm tempted to remove the ICE label. Its a little weird that we have tests that are checking the output format of ICEs, but regardless of that, this bug is *not* about addressing any ICEs in the compiler itself.)

**Comment by pnkfelix** [comment]
> Would it be enough to normalize output of those two tests similarly to this?
> 
> https://github.com/rust-lang/rust/blob/0221e265621a5fcc68ca62bdcdeabad1882a0e9a/src/test/ui/consts/miri_unleashed/mutable_references_ice.rs#L4

Yes I suspect something like that might well suffice.

**Comment by pnkfelix** [comment]
(I'm willing to mentor fixes to this bug. I suspect there are very easy solutions available.)

**Comment by hbina** [comment]
I will try to reproduce this error, if I can I want to try to take it down!

**Comment by hbina** [comment]
Tested it on Ubuntu 18.04, and it did panic. Waiting on instructions, I have little clue as how to proceed.

**Comment by cuviper** [comment]
> My assumption is that the parallel-compiler flag doesn't result in all of the threads being named `rustc`, and so when the ICE occurs it produces an error message from an unnamed thread. I also assume this is totally OK in most cases, because very few things rely on the names of compiler threads.

Yes, and I have a commit to add thread names in #64358, https://github.com/rust-lang/rust/pull/64358/commits/4152abe49492a8f95293ddd9a980948a6e080aba. We could easily apply that separately without waiting for the new rustc-rayon to be published.

**Comment by hbina** [comment]
Hmm, so I need to apply the same changes to this issue as well? @cuviper 
I might be a bit slow, I borked my Linux machine ...

**Comment by cuviper** [comment]
@hbina Yes, if you apply the same `thread_name` change, it will fix this issue. I'm also hoping that the wait on new `rustc-rayon` will be resolved soon, so we can just land my PR in full.

Either way, I don't think this issue deserves `P-high` -- the actual ICE behavior of these tests is correctly handled, but for the cosmetic thread name.

**Comment by pnkfelix** [comment]
closing as fixed since #64358 landed.
